File: Hw 5 q1/Queue.cs
using System;
//same as java and C++ code				
public class Node
{
	//data 
	public Node next;
	public String[] characterNames = new String[10];
	
	//constructor
	public Node(String[] s)
	{
		characterNames = s;
	}
	
	
}
public class queue
{
	public Node head;
	public Node tail;
	
	//empty queue (constructor)
	public queue()
	{
		head = null;
		tail = null;
	}
	//add element
	public void enequeue(String[] s)
	{
		Node hold = new Node(s);
		
		if(head == null)
		{
			head = hold;
			tail = hold;
		}
		else
		{
			tail.next = hold;
			tail= hold;
		}
		
	}
	//remove element
	public void dequeue()
	{
		if(head != null)
		{
			head = head.next;
		}
	}
	//empty list?
	public void empty()
	{
		head = null;
		tail = null;
	}
}

File: Hw 5 q1/Queue.java
class Node
{
	//10 character names
	public String characterNames[] = new String [10];
	public Node next; //refrence to last node
	
	//default constructor
	public Node()
	{
		
	}
	
	//overloaded constructor
	public Node(String[] characters)
	{
		
		for(int i=0;i<characters.length;i++)
		{
			characterNames[i] = characters[i];
		}
	
	}
}

public class Queue 
{
	
	
	private Node head;
	private Node tail;
	
	//default constructor
	public Queue()
	{
		head = null;
		tail = null;
	}
	
	
	//not sure if this is meant to empty the queue or do somthing elese, this is emptying queue
	public void empty()
	{
		head = null;
		tail = null;
	}
	
	public void enqueue(String[] s)
	{
		Node hold = new Node(s);
		//if empty head is the node
		if(head == null)
		{
			head = hold;
		}
		else
		{	//link prev tail to node
			tail.next = hold;
		}
		//set the tail to current node
		tail = hold;
	}
	
	public void dequeue()
	{
		//if empty do nothing
		if(head != null)
		{
			//if tail is equal to head, tail == null
			if(tail == head)
			{
				tail = null;
			}
			//remove head by pointing to next element of head
			else if(head.next != null)
			{
				head = head.next;
			}
		}
	}
}
